src/ui/verificationSymbolStatusView.ts

```diff
@@ -0,0 +1,318 @@
+    );
+    return instance;
+  }
+
+  private getFirstStatusForFilePromise(uriString: string): ResolveablePromise<IVerificationSymbolStatusParams> {
+    let listener = this.updateListenersPerFile.get(uriString);
+    if(listener === undefined) {
+      let storedResolve: (value: IVerificationSymbolStatusParams) => void;
+      const promise = new Promise<IVerificationSymbolStatusParams>((resolve) => {
+        storedResolve = resolve;
+      });
+      listener = { resolve: storedResolve!, promise };
+      this.updateListenersPerFile.set(uriString, listener);
+    }
+    return listener;
+  }
+
+  private constructor(
+    private readonly context: ExtensionContext,
+    private readonly languageClient: DafnyLanguageClient,
+    private readonly compilationStatusView: CompilationStatusView) {
+    this.controller = this.createController();
+  }
+
+  private itemStates: Map<string, PublishedVerificationStatus> = new Map();
+  private itemRuns: Map<string, ItemRunState> = new Map();
+  private readonly runItemsLeft: Map<TestRun, number> = new Map();
+  private readonly updateListenersPerFile: Map<string, ResolveablePromise<IVerificationSymbolStatusParams>> = new Map();
+  private readonly updatesPerFile: Map<string, IVerificationSymbolStatusParams> = new Map();
+  private readonly controller: TestController;
+  private automaticRunEnd: boolean = false;
+  private processUpdateLock: Promise<void> = Promise.resolve();
+
+  private createController(): TestController {
+    const controller = tests.createTestController('verificationStatus', 'Verification Status');
+    controller.createRunProfile('Verify', TestRunProfileKind.Run, async (request) => {
+      const items: TestItem[] = this.getItemsInRun(request, controller);
+      const runningItems: TestItem[] = [];
+      let outerResolve: () => void;
+      await this.processUpdateLock;
+      this.processUpdateLock = new Promise((resolve) => {
+        outerResolve = resolve;
+      });
+
+      const runs = items.map(item => this.languageClient.runVerification({ position: item.range!.start, textDocument: { uri: item.uri!.toString() } }));
+      for(const index in runs) {
+        const success = await runs[index];
+        if(success) {
+          runningItems.push(items[index]);
+        }
+      }
+
+      if(runningItems.length > 0) {
+        this.createRun(runningItems);
+      }
+      outerResolve!();
+    }, true);
+    return controller;
+  }
+
+  private getItemsInRun(run: TestRunRequest, controller: TestController): TestItem[] {
+    const allItems: TestItem[] = [];
+    controller.items.forEach(item => allItems.push(item));
+    const result: TestItem[] = [];
+    const todo = run.include ? [ ...run.include ] : allItems;
+    const excludes = run.exclude ? new Set(run.exclude) : new Set<TestItem>();
+    while(todo.length > 0) {
+      const current = todo.pop()!;
+      if(excludes.has(current)) {
+        continue;
+      }
+
+      current.children.forEach(child => {
```

Comment:
Is current.children ever not empty? Is the plan to nest the tests under modules, for example?